app/controllers/problems_controller.rb

```diff
@@ -12,10 +12,30 @@ def index
   end
 
   def create
-    problem = Problem.new(create_params)
-    if problem.save!
-      render status: :created, json: problem
+    problem = Problem.create!(create_params)
+
+    ActiveRecord::Base.transaction do
+      uploaded_file = params[:file]
```

Comment:
We may want more than that. Some annoying student might try to POST us a 5gb file (or even worse /dev/null). Rails must have a way of doing this, which is probably way more useful.